src/engine/SCons/Taskmaster.py

```diff
@@ -1016,6 +1016,27 @@ def will_not_build(self, nodes, node_func=lambda n: None):
                 to_visit = to_visit | parents
                 pending_children = pending_children - parents
 
+                # Clean up the other branches of side effect
+                for se in node.get_side_effects():
+                    waiting_s_e = se.get_waiting_s_e()
+                    pending_children = pending_children - waiting_s_e
+                    stack = set(waiting_s_e)
+                    if T:
+                        for n in waiting_s_e:
+                            T.write(self.trace_message('       removing side effect parent %s from the pending children set\n' %
+                                    self.trace_node(n)))
+                    while stack:
+                        n = stack.pop()
+                        waiting_parents = n.waiting_parents
+                        n.waiting_parents = set()
+                        stack = stack | waiting_parents
+                        pending_children = pending_children - waiting_parents
+
+                    for p in waiting_parents:
```

Comment:
So here, `waiting_parents` is the list from the last node in the stack above. I'm not sure, but maybe the above `while stack` loop should be accumulating _all_ the waiting parents from all nodes in the stack?